File: tools/catch_exception.rb
#!/usr/bin/ruby

# Catches if user presses CTRL-C to close program
trap 'SIGINT' do
  puts
  exit 130
end

# Checks if input is empty
def empty_input?(choice)
  return unless choice.empty?

  puts "Choice can't be empty!"
  exit
end

File: tools/catch_exception.py
#!/usr/bin/env python3

import signal
from sys import exit


# Checks if input is empty
def __input__(choice):
    if not choice or choice == '':
        print("Choice can't be empty!")
        exit()
    return choice


# Exception for python files
def handler(sig, frame):
    exit()


signal.signal(signal.SIGINT, handler)
